### PR TITLE
luarocks-nix: update 20190907 -> 20210122 

### LINKED ISSUE #110157: maintainers/scripts/update-luarocks-packages uses stdenv.lib instead of lib and urls as non-strings
I tried adding to `maintainers/scripts/luarocks-packages.csv`, then running `maintainers/scripts/update-luarocks-packages` and noticed in the diff this still uses `stdenv.lib`, and non-string URLs. 

**To Reproduce**
1. `maintainers/scripts/update-luarocks-packages pkgs/development/lua-modules/generated-packages.nix`
2. `git diff`

**Expected behavior**
The diff shouldn't contain non-string URLs, and not use `lib` via `stdenv.lib`.

**Notify maintainers**
@thoughtpolice 
@Shados 
@teto

### PR DESCRIPTION
**Pull request by teto**
###### Motivation for this change
fixes https://github.com/NixOS/nixpkgs/issues/110157

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — COMMENTED** [review]
Please follow the commit guide in the contributing guide: ``luarocks: old -> new``

**Comment by teto** [comment]
done

**Comment by teto** [comment]
actually dont merge  I need to change luarocks-nix since the output of nix-prefetch-git changed

**Comment by teto** [comment]
@ofborg  build luarocks-nix luaPackages.cassowary

**Comment by teto** [comment]
should be good now

File: pkgs/development/tools/misc/luarocks/luarocks-nix.nix [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
   src = fetchFromGitHub {
     owner = "nix-community";
     repo = "luarocks-nix";
-    rev = "73b8772e56fd39dfffda9e3b13e9eb31e93d5cde";
-    sha256 = "00jgshygw439pbaxg7yph3ijia6nid9r1br416wdbyl5wqhlhm1y";
+    rev = "v3.5.0_nix";
+    sha256 = "sha256-Ea3PVkCaUPO/mvVZtHtD1G9T/Yom28M9oN6duY4ovHk=";
   };
-  patches = [
-    ./darwin-3.1.3.patch
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/luarocks/luarocks-nix.nix:12]: Please remove this patch, too.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110467` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>lua51Packages.luarocks-nix</li>
<li>luarocks-nix (luaPackages.luarocks-nix)</li>
<li>lua53Packages.luarocks-nix</li>
<li>luajitPackages.luarocks-nix</li>
</ul>
</details>

File: pkgs/development/tools/misc/luarocks/luarocks-nix.nix (hunk 1 above)
Comment (teto) [reply pkgs/development/tools/misc/luarocks/luarocks-nix.nix:12]: it's still needed by luarocks that runs an older version

**Pull request merged**